### ISSUE TITLE
quota_manager doesn't support dimensions map argument 

### ISSUE THREAD

**Issue by vrvaderom** [terraform-google-modules/terraform-google-project-factory#665]
### TL;DR

The terraform resource google_service_usage_consumer_quota_override provides a dimensions map argument that allows configuring the quota to a subset of the available dimensions (for example, specific regions)

The quota manager does not expose this argument, and therefore does not allow setting specific dimentions of a quota override

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/service_usage_consumer_quota_override
```

### Detailed design

```markdown
The terraform resource for service_usage_consumer_quota_override defaults to no-dimension when the dimensions object argument is empty. 
```
variables.tf

```
variable "consumer_quotas" {
  description = "The quotas configuration you want to override for the project."
  type = list(object({
    service = string,
    metric  = string,
    dimensions = object({}),
    limit   = string,
    value   = string,
  }))
}
```

main.tf
```
locals {
  consumer_quotas = { for index, quota in var.consumer_quotas : "${quota.service}-${quota.metric}" => quota }
}

resource "google_service_usage_consumer_quota_override" "override" {
  provider = google-beta
  for_each = local.consumer_quotas

  project        = var.project_id
  service        = each.value.service
  metric         = each.value.metric
  limit          = each.value.limit
  dimensions     = each.value.dimensions
  override_value = each.value.value
  force          = true
}
```

### Additional information

This will *break* backwards compatibility, unless used together with 
`terraform { 
   experiments = [module_variable_optional_attrs] 
}`

and defined thus - 

```
variable "consumer_quotas" {
  description = "The quotas configuration you want to override for the project."
  type = list(object({
    service = string,
    metric  = string,
    dimensions = optional(object({})),
    limit   = string,
    value   = string,
  }))
}
```

However, optional arguments are an experimental feature of terraform which cannot be relied upon as of this moment, so it's a risky move.

**Comment by morgante** [comment]
We'd be happy to look at a PR adding this. It can be included in the next breaking release.
